sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -13,171 +13,178 @@
-      if (m_TraceEnabled)
-        printArgs(args...);
-
-      initialize();
-      auto r = m_FnPtr(args...);
-
-      if (m_TraceEnabled) {
-        std::cout << ") ---> ";
-        std::cout << (print(r),"") << "\n";
-      }
-      return r;
+
+  template <typename... Args>
+  typename std::result_of<FnType(Args...)>::type operator()(Args... args) {
+    if (m_TraceEnabled)
+      printArgs(args...);
+
+    initialize();
+    auto r = m_FnPtr(args...);
+
+    if (m_TraceEnabled) {
+      std::cout << ") ---> ";
+      std::cout << (print(r), "") << "\n";
     }
-  };
+    return r;
+  }
+};
 
-  template <typename FnType>
-  bool Trace<FnType>::m_TraceEnabled = (std::getenv("SYCL_PI_TRACE") != nullptr);
+template <typename FnType>
+bool Trace<FnType>::m_TraceEnabled = (std::getenv("SYCL_PI_TRACE") != nullptr);
 
 } // namespace pi
 
 namespace RT = cl::sycl::detail::pi;
 
-#define PI_ASSERT(cond, msg) \
-  RT::assertion((cond), "assert: " msg);
+#define PI_ASSERT(cond, msg) RT::assertion((cond), "assert: " msg);
 
 #define PI_TRACE(func) RT::Trace<decltype(func)>(func, #func)
 
 // This does the call, the trace and the check for no errors.
-#define PI_CALL(pi)                                   \
-    RT::initialize(),                                 \
-    RT::PiCall(#pi).check<cl::sycl::runtime_error>(   \
-        RT::cast<detail::RT::PiResult>(pi))
+#define PI_CALL(pi)                                                            \
+  RT::initialize(), RT::PiCall(#pi).check<cl::sycl::runtime_error>(            \
+                        RT::cast<detail::RT::PiResult>(pi))
 
 // This does the trace, the call, and returns the result
-#define PI_CALL_RESULT(pi) \
-    RT::PiCall(#pi).get(detail::RT::cast<detail::RT::PiResult>(pi))
+#define PI_CALL_RESULT(pi)                                                     \
+  RT::PiCall(#pi).get(detail::RT::cast<detail::RT::PiResult>(pi))
 
 // This does the check for no errors and possibly throws
-#define PI_CHECK(pi) \
-    RT::PiCall().check<cl::sycl::runtime_error>( \
-       RT::cast<detail::RT::PiResult>(pi))
+#define PI_CHECK(pi)                                                           \
+  RT::PiCall().check<cl::sycl::runtime_error>(                                 \
+      RT::cast<detail::RT::PiResult>(pi))
 
 // This does the check for no errors and possibly throws x
-#define PI_CHECK_THROW(pi, x) \
-    RT::PiCall().check<x>( \
-        RT::cast<detail::RT::PiResult>(pi))
+#define PI_CHECK_THROW(pi, x)                                                  \
+  RT::PiCall().check<x>(RT::cast<detail::RT::PiResult>(pi))
 
 // Want all the needed casts be explicit, do not define conversion operators.
-template<class To, class From>
-To pi::cast(From value) {
+template <class To, class From> To pi::cast(From value) {
   // TODO: see if more sanity checks are possible.
-  PI_ASSERT(sizeof(From) == sizeof(To), "cast failed size check");
+  assert(sizeof(From) == sizeof(To) && "cast failed size check");
```

Comment:
I am reverting this change. 
The call may be at place to have run time checking of the cast, and it may be valid at other locations where pi::cast is used.
In my code in pi_opencl.cpp, I am removing this call and replacing it with static_assert.
Thank you for the suggestion.